File: Chalkable.API/Exceptions/ChalkableApiException.cs
using System;

namespace Chalkable.API.Exceptions
{
    public class ChalkableApiException : Exception
    {
        public string Body { get; }

        public ChalkableApiException(string message = null, Exception e = null, string body = null) : base(message ?? string.Empty, e)
        {
            Body = body;
        }
    }

}

File: Chalkable.Web/app/chlk/lib/exception/ChalkableApiException.js
NAMESPACE('chlk.lib.exception', function () {
    "use strict";

    /** @class chlk.lib.exception.ChalkableApiException */
    EXCEPTION(
        'ChalkableApiException', [
            function $(e_) {
                BASE('ChalkableApiException exception', e_);
            }
        ]);
});